lib/rules/indent.js

```diff
@@ -1029,8 +1204,13 @@ module.exports = {
 
             WhileStatement: node => addBlocklessNodeIndent(node.body, node),
 
+            "*": setCurrentUnknownNode,
+
+            "*:exit": saveOutermostUnknownNode,
+
             "Program:exit"() {
                 addParensIndent(sourceCode.ast.tokens);
+                ignoreUnknownNodes();
```

Comment:
For performance reasons, I wonder if it would be better to do this lazily -- in other words, rather than ignoring all tokens after traversal, we could just validate tokens as normal and put an ignored-node check [here](https://github.com/eslint/eslint/blob/e52d99831efffe0efe93f5f5c440d31db0f3a057/lib/rules/indent.js#L1079), ignoring the token if necessary. That way we wouldn't need to worry about iterating over ignored tokens unless a token would actually produce indentation errors.

That said, let's focus on performance as a lower priority after we figure out the other parts of this PR.